component-tools/src/main/java/org/talend/sdk/component/tools/validator/ActionValidator.java

```diff
@@ -206,6 +213,37 @@ private Stream<String> findDiscoverSchemaExtendedErrors(final AnnotationFinder f
                 .orElseGet(Stream::empty);
     }
 
+    /**
+     * Checks method signature for @DynamicDependencies annotation.
+     * Valid signatures are:
+     * <ul>
+     * <li>public List<String> getDependencies(@Option("configuration") final TheDataset dataset)</li>
+     * </ul>
+     *
+     * @param finder
+     * @return Errors on @DynamicDependencies method
+     */
+    private Stream<String> findDynamicDependenciesErrors(final AnnotationFinder finder) {
+
+        final Stream<String> optionParameter = finder
+                .findAnnotatedMethods(DynamicDependencies.class)
+                .stream()
+                .filter(m -> !hasOption(m) || !hasTypeParameter(m, DataSet.class))
```

Comment:
I think that this won't work: `DataSet` is intended to be an annotation to a class, not extending.